services/ui/ws/window_tree.cc

```diff
@@ -900,6 +900,13 @@ void WindowTree::SendToPointerWatcher(const ui::Event& event,
                                    display_id);
 }
 
+WindowManagerState* WindowTree::AppendExternalModeWindowManagerState(
```

Comment:
OK, sounds good to me. 

I also saw the error "Main process exiting because service content_packaged_services quit unexpectedly." but I'm not sure how serious it is. Maybe we can consider this in a follow-up work.

Another concern that I had is that the std::set added keeps growing when new windows are opened and it is only destroyed once the window tree goes away (i.e. all the windows are closed). Probably we should instead destroy element when the corresponding WindowManagerDisplayRoot  goes away.

In any case, I think we can merge the changes, but keeping in mind improvements in follow-up PR.